charts/ocis/values.yaml

```diff
@@ -21,170 +47,397 @@ image:
+      #   required: true
+      # @schema
       # -- style-src directive, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src
       styleSrc:
         - "'self'"
         - "'unsafe-inline'"
 
 # Debug settings for the oCIS service
 debug:
+  # @schema
+  # type: boolean
+  # required: true
+  # @schema
   # -- Profiling enables the http://<pod>:<debug-metrics>/debug/pprof endpoint to inspect various Go runtime internals.
   # You can use the endpoint on your machine by forwarding the port, eg: `kubectl port-forward -n ocis pod/proxy-8587dc9d64-fs24l 9205:9205`
   # and then accessing the port on https://localhost:9147/debug/pprof or using the pprof command line tool: `go tool pprof -web http://localhost:9205/debug/pprof/symbol\?seconds\=10`
   profiling: false
 
-# -- Deployment strategy.
+# Deployment strategy.
 deploymentStrategy:
+  # @schema
+  # enum: [RollingUpdate, Recreate]
+  # required: true
+  # @schema
+  # -- deployment strategy type to use. Can be "RollingUpdate" or "Recreate"
   type: RollingUpdate
+  rollingUpdate:
+    # @schema
+    # type: string
+    # required: true
+    # @schema
+    # -- The maximum number of pods that can be scheduled above the desired number of pods.
+    # Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%).
+    maxSurge: 25%
+    # @schema
+    # type: string
+    # required: true
+    # @schema
+    # -- The maximum number of pods that can be unavailable during the update.
+    # Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%).
+    maxUnavailable: 25%
 
+# @schema
+# type: string
+# required: true
+# @schema
 # -- Domain where oCIS is reachable for the outside world
-externalDomain: ""
+externalDomain:
 
 # Insecure options.
 # These are useful for some limited environments like CI or on a test cluster.
 insecure:
+  # @schema
+  # type: boolean
+  # required: true
+  # @schema
   # -- Disables SSL certificate checking for connections to the openID connect identity provider.
   # Not recommended for production installations.
   oidcIdpInsecure: false
+  # @schema
+  # type: boolean
+  # required: true
+  # @schema
   # -- Disables SSL certificate checking for connections to the oCIS http apis.
   # Not recommended for production installations.
   ocisHttpApiInsecure: false
+  # @schema
+  # type: boolean
+  # required: true
+  # @schema
   # -- Disables SSL certificate checking for connections to all OCM instances
   # Not recommended for production installations.
   ocmInsecure: false
 
 cache:
+  # @schema
+  # type: string
+  # required: true
+  # @schema
   # -- Type of the cache to use.
   # Can be set to "nats-js-kv" or "redis-sentinel". There are also the non-recommended options "memory" and "noop".
   # The address of NATS / Redis Sentinel node(s) needs to be set to `cache.nodes`.
   type: "nats-js-kv"
+  # @schema
+  # type: array
+  # items:
+  #   type: string
```

Comment:
Do these `nodes` always have an `<host>:<portnummer>` nomenclature? If so we could use pattern here. Also applies for all oder nodes secure further down below.

Reply:
I think the nats client library probably has a port default, so this might also be possible without specifying a port.

Also I don't know if a protocol prefix like `tcp://` could be allowed.


I'd like to keep this vague as long as there is no strict specification from the oCIS product